### PR TITLE
[bitnami/cassandra] Add tests and publish using VIB

### PR DESCRIPTION
**Pull request by FraPazGal**
### Description of the change

The main objective of this PR is to publish our Bitnami Cassandra Helm Chart using VMware Image Builder. In order to do that, several changes are included:

- Adding the asset to the publishing workflow
- Increasing the existing test coverage of the asset by adding GOSS tests.

### Benefits

- Ensuring higher quality of the Helm charts.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could show to be potentially flaky. 

### Applicable issues

NA

### Additional information

Tested in https://github.com/FraPazGal/charts/pull/76

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/cassandra/goss/goss.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    owner: root
+  /opt/bitnami/cassandra/conf/cassandra.yaml:
+    exists: true
+    filetype: file
+    mode: "0644"
+    group: root
+    contains:
+      - /num_tokens.*{{ .Vars.cluster.numTokens }}/
+  /var/run/secrets/kubernetes.io/serviceaccount:
+    exists: {{ .Vars.serviceAccount.create }}
```
Comment (joancafom) [inline .vib/cassandra/goss/goss.yaml:15]: I think the right expression would be:

```suggestion
    exists: {{ and .Vars.serviceAccount.create .Vars.serviceAccount.automountServiceAccountToken }}
```

The serviceAccount can be created regardless of the token being mounted.

File: .vib/cassandra/goss/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    filetype: directory
+    mode: "3777"
+command:
+  {{- $user := .Vars.dbUser.user }}
+  {{- $password := .Vars.dbUser.password }}
+  {{- $nodes := .Vars.cluster.seedCount }}
```
Comment (joancafom) [inline .vib/cassandra/goss/goss.yaml:21]: I think the number of nodes != to the number of seed nodes. IIRC, the former is determined by `replicaCount`.

File: .vib/cassandra/goss/goss.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  {{- $port := .Vars.containerPorts.cql }}
+  {{- $keyspace := printf "test_%s" (randAlpha 5) }}
+  manage-keyspace:
+    exec: "cqlsh -u {{ $user }} -p {{ $password }} -e 'CREATE KEYSPACE {{ $keyspace }} WITH REPLICATION = { $$class$$: $$SimpleStrategy$$, $$replication_factor$$ : 3 };' cassandra-0.cassandra-headless {{ $port }}{{ range $e, $i := until $nodes }} && cqlsh -u {{ $user }} -p {{ $password }} -e 'USE {{ $keyspace }};' cassandra-{{ $i }}.cassandra-headless{{ end }}"
```
Comment (joancafom) [inline .vib/cassandra/goss/goss.yaml:25]: Does it make sense to have a replication_factor of `3` if we are only deploying 2 nodes?

Comment (joancafom) [inline .vib/cassandra/goss/goss.yaml:25]: I think you are missing the port in the last command:

```suggestion
    exec: "cqlsh -u {{ $user }} -p {{ $password }} -e 'CREATE KEYSPACE {{ $keyspace }} WITH REPLICATION = { $$class$$: $$SimpleStrategy$$, $$replication_factor$$ : 3 };' cassandra-0.cassandra-headless {{ $port }}{{ range $e, $i := until $nodes }} && cqlsh -u {{ $user }} -p {{ $password }} -e 'USE {{ $keyspace }};' cassandra-{{ $i }}.cassandra-headless {{ $port }}{{ end }}"
```

**Review by joancafom — COMMENTED** [review]
Thanks for your PR! I left some comments for you to review whenever you have a chance 😁

File: .vib/cassandra/goss/goss.yaml (hunk 1 above)
Comment (FraPazGal) [reply .vib/cassandra/goss/goss.yaml]: Good catch, fixed!

File: .vib/cassandra/goss/goss.yaml (hunk 2 above)
Comment (FraPazGal) [reply .vib/cassandra/goss/goss.yaml]: It is for this testing environment where I'm setting both in the `runtime_params` but I understand what you mean. Using `replicaCount` now.

File: .vib/cassandra/goss/goss.yaml (hunk 3 above)
Comment (FraPazGal) [reply .vib/cassandra/goss/goss.yaml]: It doesn't, this is a typo from a previous cluster configuration. Fixed, thanks!

Comment (FraPazGal) [reply .vib/cassandra/goss/goss.yaml]: Indeed, Cassandra uses `port = 9042` when it is not specified and I missed adding it explicitly.

**Review by agomezmoron — APPROVED** [review]
LGTM

File: .vib/cassandra/goss/goss.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    owner: root
+  /opt/bitnami/cassandra/conf/cassandra.yaml:
+    exists: true
+    filetype: file
+    mode: "0644"
+    group: root
+    contains:
+      - /num_tokens.*{{ .Vars.cluster.numTokens }}/
+  /var/run/secrets/kubernetes.io/serviceaccount:
+    exists: {{ and .Vars.serviceAccount.automountServiceAccountToken }}
```
[the remark this answers is not in the document]
Comment (FraPazGal) [reply .vib/cassandra/goss/goss.yaml]: This is a typo indeed, good catch!

**Review by joancafom — APPROVED** [review]
LGTM!

**Pull request merged**